src/test/java/pl/byczazagroda/trackexpensesappbackend/integration/FindTransactionByIDTest.java

```diff
@@ -0,0 +1,80 @@
+package pl.byczazagroda.trackexpensesappbackend.integration;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers;
+import pl.byczazagroda.trackexpensesappbackend.BaseIntegrationTestIT;
+import pl.byczazagroda.trackexpensesappbackend.exception.ErrorCode;
+import pl.byczazagroda.trackexpensesappbackend.model.FinancialTransaction;
+import pl.byczazagroda.trackexpensesappbackend.model.FinancialTransactionType;
+import pl.byczazagroda.trackexpensesappbackend.model.Wallet;
+import pl.byczazagroda.trackexpensesappbackend.repository.FinancialTransactionRepository;
+import pl.byczazagroda.trackexpensesappbackend.repository.WalletRepository;
+
+import java.math.BigDecimal;
+import java.time.Instant;
+
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+
+public class FindTransactionByIDTest extends BaseIntegrationTestIT {
+
+    @Autowired
+    FinancialTransactionRepository financialTransactionRepository;
+
+    @Autowired
+    WalletRepository walletRepository;
+
+    @BeforeEach
+    void clearDatabase() {
+        financialTransactionRepository.deleteAll();
+        walletRepository.deleteAll();
+    }
+
+    @DisplayName("Should return proper financial transaction when search Id exist in database")
+    @Test
+    public void testGetFinancialTransactionById_whenFindingTransactionWithExistingId_thenReturnFinancialTransactionWithCorrespondingId() throws Exception {
+        Wallet wallet = walletRepository.save(new Wallet("TestWallet"));
+        FinancialTransaction testFinancialTransaction = createTestFinancialTransaction(wallet, "Test1");
+        mockMvc.perform(MockMvcRequestBuilders.get("/api/transactions/{id}", testFinancialTransaction.getId())
+                        .accept(MediaType.APPLICATION_JSON))
+                .andDo(print())
```

Comment:
We need to consider it, but in my opinion this line would be great for debugging only

Reply:
Indeed, that was for debugging, to see visually what should be returned. It is not critical for the test, so if you want me to remove it I can do it